### PR TITLE
Migrate from Request to Needle

### PR DESCRIPTION
**Pull request by ThunderSon**
Needle callback functions are similar to request, so it was straightforward.

An [issue with needle](https://github.com/tomas/needle/issues/249) is that it doesn't handle relative paths, and this requires design decisions from your side.

Setting this as draft till we figure out how to set base URLs for relative paths.

### CODE REVIEW

**Comment by ThunderSon** [comment]
@NicolasMassart I did start work on this in here, if you feel like closing it and focusing on your PR that's fine by me.

**Comment by NicolasMassart** [comment]
We could just merge both PRs I think. I did not remember you created a draft when I created mine.

File: lib/proto/http.js [hunk 1]
```diff
@@ -27,14 +27,13 @@ module.exports = {
...
-            uri: encodeURI(link),
             headers: {
                 // override 'User-Agent' (some sites return `401` when the user-agent isn't a web browser)
                 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36'
             },
-            maxRedirects: 8,
-            strictSSL: false,
-            timeout: ms(timeout),
+            follow_max: 8,
+            rejectUnauthorized: false,
```
Comment (NicolasMassart) [inline lib/proto/http.js:35]: I copied that in my PR but then reading the doc a bit more I realised this was just an option passed to the Node request directly https://nodejs.org/docs/latest/api/tls.html#tls_tls_connect_options_callback

But as indicated in the doc https://nodejs.org/docs/latest/api/tls.html#tls_tls_createserver_options_secureconnectionlistener , this has no effect if requestCert is false (and it's false by default).
> rejectUnauthorized <boolean> If not false the server will reject any connection which is not authorized with the list of supplied CAs. **This option only has an effect if requestCert is true**. Default: true.

But also it's the `tls.createServer` doc so, I wonder if it exactly works the same for requests. Do you have some experience with this? Could we add a test with a self signed cert to make sure it works?

Comment (ThunderSon) [reply lib/proto/http.js:35]: Apologies, I've been a bit overwhelmed with some changes :)
I'm definitely with working on one PR as a starter.
I can dig into the above once I am a bit free, and I'll review the other PR and help.

Comment (NicolasMassart) [reply lib/proto/http.js:35]: Please, know that I totally understand how hard it is to deal with all the stuff we would like to...
I suggest we close this one for clarity and then we both work together on the other one.
Would that be fine for you?

Comment (ThunderSon) [reply lib/proto/http.js:35]: Go ahead!

**Comment by NicolasMassart** [comment]
Closed in favor of working together on #43

**Pull request closed without merging**